test/unit/auth/token-generator.spec.ts

```diff
@@ -466,4 +467,215 @@ describe('FirebaseTokenGenerator', () => {
         });
     });
   });
+
+  describe('createCustomTokenWithTenantId()', () => {
```

Comment:
These tests largely duplicate the createCustomToken() tests above, but validate that the path with tenantIds also works. Roughly half of these tests are pretty low (though not quite no) value and I'd be happy to remove them. i.e. the tests that validate the uid handling is correct, even when a tennantid is provided, e.g. 'should throw given an empty string uid'.

Opinions welcome.

Reply:
You can follow a similar approach to other tests for common API used in a tenant and non-tenant context where shared code is used. This avoids the duplication and makes it easier to make changes in the future.

Reply:
You mean like this? https://github.com/firebase/firebase-admin-node/blob/bba61e16f2bda906833c35d25e2fbc9a4c3613cc/test/unit/auth/auth.spec.ts#L244

The problem here is that:
a) We have a TokenGenerator (rather than an Auth object) and the TokenGenerator is not tenant aware.
b) The API for creating a custom token differs based on whether you want a tenantid included or not.

This problem is the more general problem of testing method overloads; I'm not aware of a great solution. Here's the obvious:
a) Just duplicate the tests. (i.e. as currently implemented)
b) Mock the implementation method and just verify that createCustomToken() pipes it's arguments through successfully. Possibly more pragmatic, but requires you to peek at the implementation (and to assume it won't change.)

In our case, we could also:
c) Make TokenGenerator tenant aware. Then we could use the same approach as for Auth instances. It seems a little odd... but I think it's workable with some refactoring around how the Auth instance creates its TokenGenerator.
d) Iterate over a list of functions, like this (which is pretty ugly):

```typescript
const createTokenFns = [
  (uid: string, tenantId: string, developerClaims?: {[key: string]: any}): Promise<string> => {
    return tokenGenerator.createCustomToken(string, developerClaims);
  }
  (uid: string, tenantId: string, developerClaims?: {[key: string]: any}): Promise<string> => {
    return tokenGenerator.createCustomTokenWithTenantId(string, tenantId, developerClaims);
  }];

createTokensFns.forEach((fn) => {
  describe("create custom token, possibly tenant aware", () => {
    it('should work', () => {
      expect(fn('uid', 'tenantid_that_might_be_ignored')).should.eventually.be.fulfilled;
    });
  });
});
```

I don't like (d); I *mildly* don't like (b). (c) doesn't seem so bad though. wdyt?

Reply:
`TokenGenerator` is created per `Auth` and `TenantAwareAuth` instance. The tenant ID in that case does not change after creation (could be set in the constructor). `TokenGenerator` has only one public method `createCustomToken` that mints tokens that are tenant aware. Every call of that should also set the tenant ID in the minted JWT. So maybe (c) is a good choice.  But I am not sure how much work is involved in this. Do you want to explore and make an assessment?

Reply:
It's not too much at all. Done.